msal-dotnet-articles/acquiring-tokens/desktop-mobile/username-password-authentication.md

```diff
@@ -1,271 +1,260 @@
 
-  ![Azure AD client types](../../media/azure-ad-client-type.png)
+>[!NOTE]
+>If your application supports authentication with personal Microsoft accounts, ROPC flow will not be available **even if** your application also supports authentication with work and school accounts.
 
-### Code
+### Sample code
 
-`PublicClientApplication`contains the method `AcquireTokenByUsernamePassword`
+ROPC flow is only available for public client applications. To use it, developers can leverage the [`PublicClientApplication`](xref:Microsoft.Identity.Client.PublicClientApplication) class, which contains the [`AcquireTokenByUsernamePassword`](xref:Microsoft.Identity.Client.AcquireTokenByUsernamePasswordParameterBuilder) method.
 
-![IPublicClientApplication interface](../../media/ipublicclientapplication-interface.png)
+The following sample showcases a simplified use-case:
 
-The following sample presents a simplified case
+>[!NOTE]
+>Replace `/contoso.com` in the authority URL with your tenant ID or `/organizations`.
 
 ```csharp
 static async Task GetATokenForGraph()
 {
- string authority = "https://login.microsoftonline.com/contoso.com";
- string[] scopes = new string[] { "user.read" };
- IPublicClientApplication app;
- app = PublicClientApplicationBuilder.Create(clientId)
-       .WithAuthority(authority)
-       .Build();
- var accounts = await app.GetAccountsAsync();
-
- AuthenticationResult result = null;
- if (accounts.Any())
- {
-  result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
-                    .ExecuteAsync();
- }
- else
- {
-  try
-  {
-   var securePassword = new string();
-   foreach (char c in "dummy")        // you should fetch the password
-    securePassword.AppendChar(c);  // keystroke by keystroke
-
-   result = await app.AcquireTokenByUsernamePassword(scopes,
-                                                    "joe@contoso.com",
-                                                     securePassword)
-                      .ExecuteAsync();
-  }
-  catch(MsalException)
-  {
-   // See details below
-  }
- }
- Console.WriteLine(result.Account.Username);
+    string authority = "https://login.microsoftonline.com/contoso.com";
+    string[] scopes = new string[]
+    {
+        "user.read"
+    };
+
+    IPublicClientApplication app;
+    app = PublicClientApplicationBuilder.Create(clientId).WithAuthority(authority).Build();
+
+    var accounts = await app.GetAccountsAsync();
+    AuthenticationResult result = null;
+
+    if (accounts.Any())
+    {
+        result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault()).ExecuteAsync();
+    }
+    else
+    {
+        try
+        {
+            var securePassword = new string();
+            foreach(char c in "dummy") // you should fetch the password
+            securePassword.AppendChar(c); // keystroke by keystroke
+            result = await app.AcquireTokenByUsernamePassword(scopes, "joe@contoso.com", securePassword).ExecuteAsync();
```

Comment:
We deprecated SecureString overload.

See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2437

```suggestion
            result = await app.AcquireTokenByUsernamePassword(scopes, "joe@contoso.com", "Password").ExecuteAsync();
```